### PR TITLE
nginx: add otel_ngx_module subpackage

### PR DESCRIPTION
**Pull request by mfrw**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
For workitem [45146424](https://microsoft.visualstudio.com/OS/_workitems/edit/45146424/)

###### Change Log  

- nginx: add otel_ngx_module
- nginx: add otel_ngx_module as a dynamic module
- nginx: add otel_ngx_module subpackage
- nginx: add %license for otel_ngx_module subpackage

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #5947
- #5913

###### Links to CVEs  
- NA

###### Test Methodology

- Local Build.
- Pipeline build: [PR-6012](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=409443&view=results)

PS: I used the new `containerized-build` feature for this and a big kudos to @neha170 for this feature
/cc @christopherco

### CODE REVIEW

File: SPECS/nginx/nginx.spec [hunk 1]
```diff
@@ -15,6 +16,7 @@ URL:            https://nginx.org/
 Source0:        https://nginx.org/download/%{name}-%{version}.tar.gz
 Source1:        nginx.service
 Source2:        https://github.com/nginx/njs/archive/refs/tags/%{njs_version}.tar.gz#/%{name}-njs-%{njs_version}.tar.gz
+Source3:        https://github.com/open-telemetry/opentelemetry-cpp-contrib/archive/%{opentelemetry_cpp_contrib_git_commit}.tar.gz#/opentelemetry-cpp-contrib-%{opentelemetry_cpp_contrib_git_commit}.tar.gz
```
Comment (christopherco) [inline SPECS/nginx/nginx.spec:19]: Wondering if any cgmanifest entry changes are needed

File: SPECS/nginx/nginx.spec [hunk 2]
```diff
@@ -39,16 +41,34 @@ The nginx-filesystem package contains the basic directory layout
...
+The OpenTelemetry module for Nginx
+
 %prep
 %autosetup -p1
 pushd ../
 mkdir nginx-njs
 tar -C nginx-njs -xf %{SOURCE2}
+mkdir otel-cpp-contrib
+tar -C otel-cpp-contrib -xf %{SOURCE3}
+# The following change is a build break in upstream and a PR has been raised to fix it.
```
Comment (christopherco) [inline SPECS/nginx/nginx.spec:62]: Can we put a comment in the spec linking to the PR?

File: SPECS/nginx/nginx.spec [hunk 3]
```diff
@@ -108,8 +128,8 @@ exit 0
 %config(noreplace) %{_sysconfdir}/%{name}/scgi_params.default
 %config(noreplace) %{_sysconfdir}/%{name}/uwsgi_params
 %config(noreplace) %{_sysconfdir}/%{name}/uwsgi_params.default
-%{_sysconfdir}/%{name}/win-utf
 %{_sysconfdir}/%{name}/html/*
+%{_sysconfdir}/%{name}/win-utf
```
Comment (christopherco) [inline SPECS/nginx/nginx.spec:132]: nit - unnecessary change

File: SPECS/nginx/nginx.spec [hunk 4]
```diff
@@ -1,12 +1,13 @@
 %global nginx_user nginx
 %global njs_version 0.7.12
+%global opentelemetry_cpp_contrib_git_commit 37e4466d882cbddff6f607a20fe327060de76166
```
Comment (christopherco) [inline SPECS/nginx/nginx.spec:3]: Curious how this git commit was chosen. Does it correspond to a specific tag/release?

File: SPECS/nginx/nginx.spec (hunk 3 above)
Comment (mfrw) [reply SPECS/nginx/nginx.spec:132]: Ahh this must have been the `sort` when i did it ..

File: SPECS/nginx/nginx.spec (hunk 4 above)
Comment (mfrw) [reply SPECS/nginx/nginx.spec:3]: This is the latest `master`.
The project does not follow a good release name pattern unfortunately.

File: SPECS/nginx/nginx.spec (hunk 2 above)
Comment (mfrw) [reply SPECS/nginx/nginx.spec:62]: Done

File: SPECS/nginx/nginx.spec (hunk 1 above)
Comment (mfrw) [reply SPECS/nginx/nginx.spec:19]: Yes, I was un-sure about this, will add. Thank you for the suggestion.

File: SPECS/nginx/nginx.spec [hunk 5]
```diff
@@ -39,16 +41,35 @@ The nginx-filesystem package contains the basic directory layout
 for the Nginx server including the correct permissions for the
 directories.
 
+%package otel_ngx_module
+License:        Apache-2.0
```
Comment (kanikanema) [inline SPECS/nginx/nginx.spec:45]: Curious, this is a different license from the main package. How does that work? Does this break any of our license checking tools?

**Review by kanikanema — COMMENTED** [review]
LGTM

**Review by kanikanema — APPROVED** [review]
